iree/hal/cts/executable_cache_test.h

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_HAL_CTS_EXECUTABLE_CACHE_TEST_H_
+#define IREE_HAL_CTS_EXECUTABLE_CACHE_TEST_H_
+
+#include "iree/base/api.h"
+#include "iree/base/string_view.h"
+#include "iree/hal/api.h"
+#include "iree/hal/cts/cts_test_base.h"
+#include "iree/testing/gtest.h"
+#include "iree/testing/status_matchers.h"
+
+// TODO(scotttodd): include a header for this (shared with generate_embed_data)?
+#ifndef IREE_FILE_TOC
+#define IREE_FILE_TOC
+
+extern "C" {
+typedef struct iree_file_toc_t {
+  const char* name;
+  const char* data;
+  size_t size;
+} iree_file_toc_t;
+}
+
+#endif  // IREE_FILE_TOC
+
+extern "C" {
+const iree_file_toc_t* iree_cts_testdata_abs_create();
+}
```

Comment:
Another idea for this: only declare a `char* data, size_t size` (or just `iree_const_byte_span_t`) interface here in the templated test header file, then implement that interface using `iree_file_toc_t` and `c_embed_data` in the source files.

Then `cts_test_template.cc` would have something like

```c++
#include "path/to/generated/header.h"

iree_const_byte_span_t get_executable_data_abs() {
  auto* toc = iree_cts_testdata_abs_create();
  return iree_make_const_byte_span(toc->data, toc->size);
}

// One of those for each test executable (one with push constants, etc.)
```

(data ownership should be made more clear though)

Reply:
I'm liking this idea - may add a commit for it after working through my code review queue...

Reply:
Pushed that commit. Not super clean, but better than before IMO...